plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -397,4 +408,16 @@ private IcebergSplit toIcebergSplit(FileScanTask task)
                         .collect(toImmutableList()),
                 SplitWeight.fromProportion(Math.min(Math.max((double) task.length() / tableScan.targetSplitSize(), minimumAssignedSplitWeight), 1.0)));
     }
+
+    private static Domain getPathDomain(TupleDomain<IcebergColumnHandle> effectivePredicate)
+    {
+        return effectivePredicate.getDomains()
+                .flatMap(domains -> Optional.ofNullable(domains.get(pathColumnHandle())))
+                .orElse(Domain.all(pathColumnHandle().getType()));
```

Comment:
if effective predicate is none, this method returns ALL domain.
The NONE tuple domain needs to be handled explicitly.

Reply:
OTOH, the none is not expected here (should be filtered out earlier), so sth like
```java
IcebergColumnHandle pathColumn = pathColumnHandle();
Domain domain = effectivePredicate.getDomains().orElseThrow(() -> new IllegalArgumentException("Unexpected NONE tuple domain"))
        .get(pathColumn);
if (domain == null) {
    return Domain.all(pathColumn.getType());
}
return domain;
```